src/app/shared/validators/email-validator.directive.ts

```diff
@@ -0,0 +1,28 @@
+import { Directive } from '@angular/core';
+import { Validator, NG_VALIDATORS, FormControl } from '@angular/forms';
+
+@Directive({
+  selector: '[appEmailValidator]',
+  providers: [
+    {
+      provide: NG_VALIDATORS,
+      useExisting: EmailValidatorDirective,
+      multi: true,
+    },
+  ],
+})
+export class EmailValidatorDirective implements Validator {
+  constructor() {}
+
+  validate(control: FormControl) {
```

Comment:
if extract validate function from declaration, it can be used as validator function in reactive form 
https://angular.io/guide/form-validation#adding-cross-validation-to-reactive-forms